apps/meteor/app/livechat/imports/server/rest/rooms.ts

```diff
@@ -70,3 +71,34 @@ API.v1.addRoute(
 		},
 	},
 );
+
+API.v1.addRoute(
+	'livechat/rooms/:rid/priority',
+	{
+		authRequired: true,
+		validateParams: { POST: isPOSTLivechatRoomPriorityParams },
+		permissionsRequired: {
+			POST: { permissions: ['view-l-room'], operation: 'hasAny' },
+			DELETE: { permissions: ['view-l-room'], operation: 'hasAny' },
+		},
+	},
+	{
+		async post() {
+			const { rid } = this.urlParams;
+			const { priorityId } = this.bodyParams;
+			const result = LivechatRooms.setPriorityByRoomId(rid, priorityId);
+			if (!result) {
+				return API.v1.failure({ error: 'Error setting priority' });
+			}
+			return API.v1.success({ roomId: rid, priorityId });
+		},
+		async delete() {
+			const { rid } = this.urlParams;
+			const result = LivechatRooms.unsetPriorityByRoomId(rid);
```

Comment:
and here too - missing `await` :eyes:

Reply:
it is, thanks for spotting it :see_no_evil: